src/utils/useGlobalState.ts

```diff
@@ -0,0 +1,10 @@
+import { create } from "zustand";
+
+interface GlobalState {
+  selectedBranchId?: string;
+  setSelectedBranchId: (branchId: string) => void;
+}
+
+export const useGlobalState = create<GlobalState>((set) => ({
+  setSelectedBranchId: (branchId) => set({ selectedBranchId: branchId }),
```

Comment:
```suggestion
  selectBranch: (branchId) => set({ selectedBranchId: branchId }),
```
I would prefer if we could use less generic language for the dispatchers

Reply:
Dun